third_party/mkl/build_defs.bzl

```diff
@@ -77,6 +78,7 @@ def if_enable_mkl(if_true, if_false = []):
     """
     return select({
         "@org_tensorflow//third_party/mkl:enable_mkl": if_true,
+        "@org_tensorflow//third_party/mkl:build_with_mkl_aarch64": if_true,
```

Comment:
This probably will result in a error if both the conditions are true. I don't think you need to check for build_with_mkl_aarch64 here since enable_mkl will be true.

Reply:
It indeed may be an overkill here, thanks.

Reply:
Hi @agramesh1 , the reason for adding it here was that the enable_mkl() macro depends on enable=true and build_with_mkl=true and _I believe_ it is set to true if both conditions are set to true (https://github.com/tensorflow/tensorflow/blob/fe7c743b4734b29bc7fe12b3b09b4e8de4ecb5d4/third_party/mkl/BUILD#L56) . In our builds, we found that this that enable_mkl() is not activated under these conditions which means that even if we get a successful build with oneDNN it does pick oneDNN kernels in the verbose logs when running benchmarks.  The dependency on build_with_mkl in our case is false because we define a build_with_mkl_aarch64==true instead in .bazelrc . We can either re-define enable_mkl() here to activate with build_with_mkl_aarch64==true but that would again will work only if all three conditions are true which won't be the case. I suggest having a different enable_mkl_aarch64() instead. What do you think?

```
config_setting(
    name = "enable_mkl",
    define_values = {
        "enable_mkl": "true",
        "build_with_mkl": "true",
    },
    visibility = ["//visibility:public"],
)

```

In .bazelrc
```
build:mkl_aarch64 --define=build_with_mkl_aarch64=true --define=enable_mkl=true
build:mkl_aarch64 --define=tensorflow_mkldnn_contraction_kernel=0
build:mkl_aarch64 --define=build_with_mkl_opensource=true
```

Reply:
The line 
`"@org_tensorflow//third_party/mkl:build_with_mkl_aarch64": if_true,`
is indeed not needed here, we have this check earlier in this file in `if_mkl(if_true, if_false = [])`. I have tested, and everything actually builds fine without this check.

Reply:
@cfRod and @alenik01 ok, you can leave it as it is for now.  As you can see we are restructuring the code now.
We will have to revisit this again soon.  There is a lot of legacy code leftover from a while ago and the names may no longer make sense, we plan to clean the build code shortly.

Reply:
@agramesh1, many thanks for the heads up, cleaning the build code will simplify the life significantly.